src/mount/direntry_cache.h

```diff
@@ -544,23 +607,38 @@ class DirEntryCache {
 		fifo_list_.push_back(entry);
 		entry.timestamp = timestamp;
 		entry.attr = de.attributes;
+		entry.next_index = de.next_index;
 	}
 
-	IndexSet::iterator addEntry(const SaunaClient::Context &ctx, uint32_t parent_inode,
-	                            uint32_t inode, uint64_t index, uint64_t next_index,
-								std::string name, Attributes attr, uint64_t timestamp) {
-		DirEntry *entry =
-		        new DirEntry(ctx, parent_inode, inode, index, next_index, name, attr, timestamp);
-		lookup_set_.insert(*entry);
-		auto result = index_set_.insert(*entry);
+	void addEntry(const SaunaClient::Context &ctx, uint32_t parent_inode,
+	              uint32_t inode, uint64_t index, uint64_t next_index,
+	              std::string name, Attributes attr, uint64_t timestamp) {
+		DirEntry *entry = new DirEntry(ctx, parent_inode, inode, index,
+		                               next_index, name, attr, timestamp);
+		if (parent_inode != INVALID_PARENT && !name.empty()) {
+			lookup_set_.insert(*entry);
+		}
```

Comment:
Consider checking entry was properly allocated before inserting it to avoid possible crash of the system.